src/main/js/controls/Timeline/Timeline.js

```diff
@@ -197,6 +197,20 @@ class Timeline extends Construct {
             );
         createDefs(this.config, this.svg);
         createAxes(this.axis, this.scale, this.config, this.svg);
+        const ticks = document.getElementsByClassName("tick");
+        const firstTick = ticks[0].getBoundingClientRect().width;
+        const lastTick = ticks[ticks.length - 1].getBoundingClientRect().width;
+        const xAxis = document
+            .getElementsByClassName("carbon-axis carbon-axis-x")[0]
+            .getBoundingClientRect().width;
+        if (xAxis + firstTick / 2 + lastTick / 2 > this.config.canvasWidth) {
+            d3RemoveElement(this.graphContainer, "defs");
```

Comment:
Why delete defs when you are creating it in line 198?

Reply:
I am deleting both the X axis and defs and then re creating them with the new padding values, because deleting only the X axis gives console errors.